lib/sdk/writer/index.js

```diff
@@ -515,6 +515,12 @@ class ImageWriter extends EventEmitter {
       const pipeline = this._createVerifyPipeline(destination)
 
       pipeline.on('error', (error) => {
+        // NOTE: As the `blockmap` module doesn't set our custom error codes,
+        // we need to patch `EVALIDATION` into a range checksum error here
+        if (error.message && error.message.startsWith('Invalid checksum for range')) {
+          error.code = 'EVALIDATION'
+          this.emit('fail', { device: destination.device.device, error })
```

Comment:
Is it intended to emit both `error` and `fail` events ?

Reply:
That's something we've discussed last evening briefly; it shouldn't cause any trouble, but we need to test this to be sure.